actors/common/src/main/java/org/sunbird/common/ShadowUserProcessor.java

```diff
@@ -0,0 +1,361 @@
+            registerUserToOrg(userId,orgId);
+        }
+        syncUserToES(userId);
+        updateUserInShadowDb(userId,shadowUser,ClaimStatus.CLAIMED.getValue());
+    }
+
+    private boolean isRootOrgMatchedWithOrgId(String rootOrgId,String orgId){
+        if(StringUtils.equalsIgnoreCase(rootOrgId,orgId)){
+            return true;
+        }
+        return false;
+    }
+    private  void updateOrganisationsOfUsers(ShadowUser shadowUser,String rootOrgId,List<Map<String,Object>>organisations){
+        organisations.stream().forEach(organisation->{
+            String orgId=(String) organisation.get(JsonKey.ORGANISATION_ID);
+            if(!isRootOrgMatchedWithOrgId(rootOrgId,orgId)){
+                String id=(String) organisation.get(JsonKey.ID);
+                updateStatusInUserOrg(shadowUser,id);
+            }
+        });
+    }
+    private void updateStatusInUserOrg(ShadowUser shadowUser,String id){
+        Map<String,Object>propertiesMap=new HashMap<>();
+        propertiesMap.put(JsonKey.ID,id);
+        propertiesMap.put(JsonKey.IS_DELETED,true);
+        propertiesMap.put(JsonKey.UPDATED_BY,shadowUser.getAddedBy());
+        propertiesMap.put(JsonKey.UPDATED_DATE,ProjectUtil.getFormattedDate());
+        Response response=cassandraOperation.updateRecord(JsonKey.SUNBIRD,JsonKey.USER_ORG,propertiesMap);
+        ProjectLogger.log("ShadowUserProcessor:updateStatusInUserOrg:response from cassandra in updating user org ".concat(response.getResult()+""));
+    }
+
+    private List<Map<String, Object>> getUserFromEs(ShadowUser shadowUser) {
+        Map<String, Object> request = new HashMap<>();
+        Map<String, Object> filters = new HashMap<>();
+        Map<String, Object> or = new HashMap<>();
+        or.put(JsonKey.EMAIL, StringUtils.isNotBlank(shadowUser.getEmail()) ? getEncryptedValue(shadowUser.getEmail()) : "");
+        or.put(JsonKey.PHONE, StringUtils.isNotBlank(shadowUser.getPhone()) ? getEncryptedValue(shadowUser.getPhone()) : "");
+        filters.put(JsonKey.ES_OR_OPERATION, or);
+        filters.put(JsonKey.ROOT_ORG_ID, getCustodianOrgId());
+        request.put(JsonKey.FILTERS, filters);
+        SearchDTO searchDTO = ElasticSearchHelper.createSearchDTO(request);
+        Map<String, Object> response = (Map<String, Object>) ElasticSearchHelper.getResponseFromFuture(elasticSearchService.search(searchDTO, JsonKey.USER));
+        return (List<Map<String, Object>>) response.get(JsonKey.CONTENT);
+    }
+
+    private void updateUser(ShadowUser shadowUser) {
+        List<Map<String, Object>> esUser = getUserFromEs(shadowUser);
+        if (CollectionUtils.isNotEmpty(esUser)) {
+            if (esUser.size() == 1) {
+                Map<String, Object> userMap = esUser.get(0);
+                if (!isSame(shadowUser, userMap)) {
+                    String rootOrgId = getRootOrgIdFromChannel(shadowUser.getChannel());
+                    if (StringUtils.isEmpty(rootOrgId)) {
```

Comment:
not Required:
if (StringUtils.isEmpty(rootOrgId)) {
                        updateUserInShadowDb(null, shadowUser, ClaimStatus.REJECTED.getValue());
                        return;
                    }

Reply:
done